projects/plugins/jetpack/tests/php/modules/search/test-class-jetpack-instant-search.php

```diff
@@ -64,13 +58,84 @@ public function test_remove_wp_migrated_widgets() {
 			'sidebar-2'                      => array(),
 			'array_version'                  => 3,
 		);
-
-		add_filter( 'option_sidebars_widgets', array( static::class, 'getOldSidebarsWidgets' ) );
-		static::$instant_search->save_old_sidebars_widgets( $old_sidebars_widgets );
+		$this->set_private_member_value( static::$instant_search, 'old_sidebars_widgets', $old_sidebars_widgets );
 
 		$this->assertEquals(
 			$expected_sidebars_widgets,
 			static::$instant_search->remove_wp_migrated_widgets( $new_sidebars_widgets )
 		);
 	}
+
+	/**
+	 * Can set old_sidebars_widgets value when _wp_sidebars_changed action is set
+	 */
+	public function test_save_old_sidebars_widgets_with__wp_sidebars_changed() {
+		// Set old_sidebars_widgets to null.
+		$this->set_private_member_value( static::$instant_search, 'old_sidebars_widgets' );
+		$old_sidebars_widgets = $this->get_old_sidebars_widgets_data();
+		static::$instant_search->save_old_sidebars_widgets( $old_sidebars_widgets );
+
+		$this->assertEquals( $old_sidebars_widgets, $this->get_private_member_value( static::$instant_search, 'old_sidebars_widgets' ) );
+	}
+
+	/**
+	 * Can not set old_sidebars_widgets value when _wp_sidebars_changed action is set
+	 */
+	public function test_save_old_sidebars_widgets_with_no__wp_sidebars_changed() {
+		// Set old_sidebars_widgets to null.
+		$this->set_private_member_value( static::$instant_search, 'old_sidebars_widgets' );
+		$old_sidebars_widgets = $this->get_old_sidebars_widgets_data();
+		remove_action( 'after_switch_theme', '_wp_sidebars_changed' );
+		static::$instant_search->save_old_sidebars_widgets( $old_sidebars_widgets );
+
+		$this->assertNull( $this->get_private_member_value( static::$instant_search, 'old_sidebars_widgets' ) );
+	}
+
+	/**
+	 * Mocked sidebars_widgets data
+	 *
+	 * @since 9.9.0
+	 */
+	private function get_old_sidebars_widgets_data() {
```

Comment:
```suggestion
	private function get_old_sidebars_widgets_fixture() {
```

Clarifying that the `data` here is meant to be only used within the testing environment as a `fixture` :)